File: src/main/java/jisop/domain/Argument.java
package jisop.domain;

import java.util.Objects;
import java.util.function.Consumer;

/**
 * Created by mathieu.
 */
public class Argument {
    public final String description;
    public final String name;
    public final Consumer<String> action;
    public final boolean required;
    public final Class type;
    public Argument(String name, Consumer<String> action, boolean required, String description ,Class type)
    {
        this.description = description;
        this.name = name;
        this.action = action;
        this.required = required;
        this.type = type;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 79 * hash + Objects.hashCode(this.description);
        hash = 79 * hash + Objects.hashCode(this.name);
        hash = 79 * hash + Objects.hashCode(this.required);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Argument other = (Argument) obj;
        return Objects.equals(this.description, other.description) &&
                Objects.equals(this.name, other.name) &&
                Objects.equals(this.required, other.required);
    }
    public String toString(){
        return String.format("%s:%s", name, description);
    }
}

File: src/main/java/isop/domain/Argument.kt
package isop.domain

import java.util.Objects
import java.util.function.Consumer

/**
 * Created by mathieu.
 */
open class Argument(val name: String, val action: Consumer<String>?, val required: Boolean, val description: String?) {

    override fun hashCode(): Int {
        var hash = 7
        hash = 79 * hash + Objects.hashCode(this.description)
        hash = 79 * hash + Objects.hashCode(this.name)
        hash = 79 * hash + Objects.hashCode(this.required)
        return hash
    }

    override fun equals(obj: Any?): Boolean {
        if (obj == null) {
            return false
        }
        if (javaClass != obj.javaClass) {
            return false
        }
        val other = obj as Argument?
        return this.description == other!!.description &&
                this.name == other.name &&
                this.required == other.required
    }

    override fun toString(): String {
        return String.format("%s:%s", name, description)
    }
}
